File: src/demos/ps/dungeons/Baya_cave.java
package demos.ps.dungeons;

import static core.Script.current_map;
import demos.ps.PSDungeon;
import demos.ps.oo.PSGame;
import demos.ps.oo.PSGame.Planet;

public class Baya_cave {
	
	public static void startmap() {
		PSDungeon dungeon = PSGame.currentDungeon;
		dungeon.startDungeon();
	}
	
	public static void drmad() {
		// Nonsense: Dr.Mad is not here anymore
	}
	public static void exit() {
		PSGame.mapswitch(Planet.PALMA,121,16);
	}
	public static void nextArea() {
		PSGame.mapswitch(Planet.PALMA,116,9);
	}

}

File: src/demos/ps/dungeons/Baya_cave.ts
/**
 * Baya_cave - Dungeon Script
 * TypeScript port of Baya_cave.java
 */

import { PSGame } from '../PSGame';
import { Planet } from '../game/City';
import { MainEngine } from '../../../core/MainEngine';

export class Baya_cave {

  public static async startmap(): Promise<void> {
    MainEngine.setScriptContext(Baya_cave);
    const dungeon = PSGame.getCurrentDungeonInstance();
    if (dungeon) {
      await dungeon.startDungeon();
    }
  }

  public static drmad(): void {
    // Nonsense: Dr.Mad is not here anymore (Java comment)
  }

  public static async exit(): Promise<void> {
    await PSGame.mapswitchToPlanet(Planet.PALMA, 121, 16);
  }

  public static async nextArea(): Promise<void> {
    await PSGame.mapswitchToPlanet(Planet.PALMA, 116, 9);
  }
}
